shape.h

```diff
@@ -0,0 +1,34 @@
+#ifndef SHAPE_H
+#define SHAPE_H
+#include "point.h"
+
+class Shape
+{
+    virtual bool contains(const Point & p) const = 0;
+};
+
+class Ellipse: public Shape //This must be override pure virtual function
+{
+    public:
+        Ellipse(Point(10-(f/3),35+(f/3)), Point(25-(f/3),35+(f/3)), 22+f );
```

Comment:
must also include a prototype for `contains`, since it is pure virtual in the parent class

also, shouldn't put expressions inside the constructor prototype